### PR TITLE
SDK-832 refactor workflow listener

### PR DESCRIPTION
**Pull request by sergeyspryker**
https://spryker.atlassian.net/browse/SDK-832

### CODE REVIEW

File: src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php [hunk 1]
```diff
@@ -207,9 +206,9 @@ protected function tryRunTask(TransitionEvent $event, WorkflowTransitionInterfac
     protected function resolverNextTransition(TransitionEvent $event, ContextInterface $context): ?string
     {
         $transitionResolverService = $this->getTransitionMeta($event, static::META_KEY_TRANSITION_RESOLVER);
-        if ($transitionResolverService && isset($transitionResolverService['service'])) {
-            $transitionResolver = $this->container->get($transitionResolverService['service']);
-            if ($transitionResolver instanceof TransitionResolverInterface) {
+        if ($transitionResolverService && isset($transitionResolverService['name'])) {
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php:209]: use early return to avoid unneccessary nesting

File: docker-compose.yml [hunk 2]
```diff
@@ -2,28 +2,23 @@ version: '3'
...
       - $PWD:/project:rw
-      - ./db:/data/db:rw
-      - ./var:/data/var:rw
-      - ./extension:/data/extension:rw
-      - ./VERSION:/data/VERSION:rw
-      - ./config/packages/workflow.yaml:/data/config/packages/workflow.yaml:rw
-      - ~/.ssh:/home/spryker/.ssh:ro
-      - ./.gitmodules:/data/.gitmodules:rw
+      - .:/data:rw
+#      - ./db:/data/db:rw
```
Comment (pavelmaksimov25) [inline docker-compose.yml:19]: Please avoid single-line comments. We don't need them as long as we have git history.

File: tests/Sdk/Unit/Infrastructure/Service/WorkflowTransitionResolverRegistryTest.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+use Codeception\Test\Unit;
+use SprykerSdk\Sdk\Infrastructure\Service\WorkflowTransitionResolverRegistry;
+use SprykerSdk\SdkContracts\Workflow\TransitionResolverInterface;
+
+class WorkflowTransitionResolverRegistryTest extends Unit
+{
+    /**
+     * @return void
+     */
+    public function testReturnWorkflowTransitionResolverWhenItSet(): void
```
Comment (pavelmaksimov25) [inline tests/Sdk/Unit/Infrastructure/Service/WorkflowTransitionResolverRegistryTest.php:19]: please add test for a negative case

File: src/Infrastructure/Service/WorkflowTransitionResolverRegistry.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+{
+    /**
+     * @var iterable<\SprykerSdk\SdkContracts\Workflow\TransitionResolverInterface>
+     */
+    protected iterable $workflowTransitionResolvers;
+
+    /**
+     * @var array<string, \SprykerSdk\SdkContracts\Workflow\TransitionResolverInterface>|null
+     */
+    protected ?array $transitionResolverMap = null;
```
Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Service/WorkflowTransitionResolverRegistry.php:22]: Does it make sense to have nullable array here?
Let's just put `protected array $transitionResolverMap = [];`

**Review by pavelmaksimov25 — APPROVED** [review]

File: src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php [hunk 5]
```diff
@@ -207,14 +206,16 @@ protected function tryRunTask(TransitionEvent $event, WorkflowTransitionInterfac
     protected function resolverNextTransition(TransitionEvent $event, ContextInterface $context): ?string
     {
         $transitionResolverService = $this->getTransitionMeta($event, static::META_KEY_TRANSITION_RESOLVER);
-        if ($transitionResolverService && isset($transitionResolverService['service'])) {
-            $transitionResolver = $this->container->get($transitionResolverService['service']);
-            if ($transitionResolver instanceof TransitionResolverInterface) {
-                return $transitionResolver->resolveTransition($context, $transitionResolverService['settings']);
-            }
+        if (!$transitionResolverService || !isset($transitionResolverService['name'])) {
```
Comment (vol4onok) [inline src/Infrastructure/Event/Workflow/WorkflowTransitionListener.php:209]: it's enough `if(!isset($transitionResolverService['name']))`

**Review by vol4onok — APPROVED** [review]

File: src/Extension/Workflow/TransitionBooleanResolver.php [hunk 6]
```diff
@@ -22,6 +22,19 @@ class TransitionBooleanResolver implements TransitionResolverInterface
      */
     public const FAILED = 'failed';
 
+    /**
+     * @var string
+     */
+    protected const RESOLVER_NAME = 'boolean_resolver';
```
Comment (vol4onok) [inline src/Extension/Workflow/TransitionBooleanResolver.php:28]: you need to fix it here  if you resolve it by name
image

File: src/Infrastructure/Service/WorkflowTransitionResolverRegistry.php (hunk 4 above)
Comment (sergeyspryker) [reply src/Infrastructure/Service/WorkflowTransitionResolverRegistry.php:22]: How can i check the registry has been populated already?

**Review by vol4onok — APPROVED** [review]

**Review by pavelmaksimov25 — APPROVED** [review]
